sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -126,7 +126,10 @@ private boolean shouldApplyScopeData(
       return SentryId.EMPTY_ID;
     }
 
-    Session session = null;
+    @Nullable
+    Session oldSession =
```

Comment:
Is this the only way to get a snapshot of the session before updating?

Reply:
`val session = it.withSession { }`

Most likely since the session methods should not be public, it's internal to the SDK only.